### ISSUE TITLE
Changing source from Any source to Filesystem closes and reopens neo-tree instead of switching source

### ISSUE THREAD

**Issue by braxtons12** [nvim-neo-tree/neo-tree.nvim#713]
Title says it all, switching source, either by clicking the source_selector or pressing "<" or ">", from one source (eg git_status), back to filesystem causes neo-tree to close then reopen on the filesystem source, instead of just switching to filesystem. This causes a jarring flash of the main window growing to the left then shrinking back when neo-tree reopens. This all occurs in ~1 second (or less).

**Comment by pysan3** [comment]
I am able to reproduce this but not sure why this is happening.

I will dig into this since it might be related to the source selector.

**Comment by pysan3** [comment]
So, I now found out what the cause is. (commenting mainly for my memo)

For all sources, the logic is that the window for the previous source is closed [here](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8238865e1d9c61f1a260c290653f2c419503e0a9/lua/neo-tree/command/init.lua#L161) and then a new window for the next source is calculated afterwards.

> This causes a jarring flash of the main window.

So this is actually a _feature_ and not a bug now.

The reason why we cannot observe a flash on buffers / git source etc is that the new buffer is calculated synchronously and therefore, all happens **before** any rendering update of neovim.
(Technically, there is no `vim.schedule` or other async functions between the previous delete and the spawning of new tree)

So, @branxtos12, if you really hate seeing the flickering, an urgent fix would be to set these options in config to make everything calculated in sync, just like other sources.
However, I would not recommend this setting because (obviously) it does everything synchronously meaning Neotree would require time to show up and neovim completely hangs during that period (which is but like half second so might be better than having flickering? Up to you).

```lua
config = {
  git_status_async = false,
  filesystem = {
    async_directory_scan = "never",
  },
}
```

I believe we can delay the deletion of previous buffer, hence the window, until the calculation is done and shift the deletion to the same rendering frame of creating the new window to avoid this issue.

I will try to implement a solution and create a PR.

**Comment by charbelnicolas** [comment]
Ohh yes, I've been noticing this behavior recently too and the flicker is a bit annoying

**Comment by charbelnicolas** [comment]
https://user-images.githubusercontent.com/14960414/218556272-809e5ba1-6985-4d37-a663-4bcb62ed46cf.mp4

**Comment by pysan3** [comment]
Hi, I have created a PR aiming to solve this issue at #752.

I would be more than happy if you could test it out with your default configs.
And also would like to hear your feedbacks. Mainly about the performance.

Thanks in advance ;)

**Comment by cseickel** [comment]
I think this is now fixed in `main` with #1011
